Mathlib/SetTheory/Ordinal/CantorNormalForm.lean

```diff
@@ -65,80 +63,132 @@ We special-case `CNF 0 o = CNF 1 o = [(0, o)]` for `o ≠ 0`.
 `CNF b (b ^ u₁ * v₁ + b ^ u₂ * v₂) = [(u₁, v₁), (u₂, v₂)]` -/
 @[pp_nodot]
 def CNF (b o : Ordinal) : List (Ordinal × Ordinal) :=
-  CNFRec b [] (fun o _ho IH ↦ (log b o, o / b ^ log b o)::IH) o
+  CNFRec b [] (fun o _ IH ↦ (log b o, o / b ^ log b o)::IH) o
+
+/-- The exponents of the Cantor normal form are stored in the first entries. -/
+def CNF.exponents (b o : Ordinal) := (CNF b o).map Prod.fst
```

Comment:
Honestly, I think putting everything here in the `CNF` namespace is not a bad idea. Otherwise we end up with `CNF_exponents` and `CNF_coeffs` and later down the line `CNF_coeff` too - write that down across every single theorem name and it gets unwieldy fast.

Reply:
You can think of this as matching `Polynomial` in some sense - we have `Polynomial.coeff` rather than `polynomialCoeff` or anything of the sort. This also comes with the added benefit that results on the CNF won't clutter the ordinal namespace for anyone not interested in the CNF.

Reply:
I've pushed a commit that namespaces most of the file within the `CNF` namespace. I think it's a nice improvement to having to type `CNF_` everywhere, but let me know what you think.

Reply:
A funny consequence of this is that basically every single theorem that existed in this file has to be deprecated... well, it's not like anyone was using this API anyways :P

Reply:
If this is a change you agree with, I'll do the namespacing in yet another separate PR. I'll stop rambling now :)

Reply:
That's a good point about cluttering `Ordinal`. This approach looks good!

Reply:
Oh actually, it doesn't make much sense to do the namespacing in a separate PR, since that would introduce yet another round of deprecations. I'll just keep this PR as is.